### PR TITLE
Stabilize `cargo logout`

### LINKED ISSUE #11884: `cargo help logout` suggests running itself
### Problem

The output of `cargo help logout` ends with “Run `cargo help logout` for more detailed information.”, but that was already what I was running. Running it again gives the same information again, not more detailed.

### Steps

Run `cargo +nightly help logout`. Output:

```
Remove an API token from the registry locally

Usage: cargo logout [OPTIONS]

Options:
  -q, --quiet                Do not print cargo log messages
      --registry <REGISTRY>  Registry to use
  -v, --verbose...           Use verbose output (-vv very verbose/build.rs output)
      --color <WHEN>         Coloring: auto, always, never
      --frozen               Require Cargo.lock and cache are up to date
      --locked               Require Cargo.lock is up to date
      --offline              Run without accessing the network
      --config <KEY=VALUE>   Override a configuration value
  -Z <FLAG>                  Unstable (nightly-only) flags to Cargo, see 'cargo -Z help' for details
  -h, --help                 Print help

Run `cargo help logout` for more detailed information.
```

### Possible Solution(s)

For logout in particular, make `cargo help logout` actually print more information. The information I was looking for in particular is: does `logout` remove `$CARGO_HOME/credentials.toml`, does it revoke the token server-side, or both?

In general, in case this does or could happen for other (or future new) commands: when no detailed help page is available, having `cargo help $COMMAND` default to printing the same as `cargo $COMMAND --help` instead is fine. But in that case neither command should suggest running `cargo help $COMMAND` for more information that isn’t there.

### Notes

_No response_

### Version

```text
cargo 1.70.0-nightly (15d090969 2023-03-21)
release: 1.70.0-nightly
commit-hash: 15d090969743630bff549a1b068bcaa8174e5ee3
commit-date: 2023-03-21
host: aarch64-unknown-linux-gnu
libgit2: 1.5.0 (sys:0.16.0 vendored)
libcurl: 8.0.1-DEV (sys:0.4.61+curl-8.0.1 vendored ssl:OpenSSL/1.1.1q)
os: Ubuntu 22.04 (jammy) [64-bit]
```

### LINKED ISSUE #11910: Stabilize `cargo logout`
This is a proposal to stabilize the `cargo logout` command.

This command is a fairly straightforward companion to `cargo login`. It will remove the token for a registry from `~/.cargo/credentials.toml`, or from a credential store with `-Zcredential-process`. This was added as part of `-Zcredential-process` because otherwise it would be quite awkward to try to remove a credential. However, it also has utility for the traditional `credentials.toml` file in the same way that `cargo login` is helpful. A user can manually edit the file in both cases, but providing a CLI command should be more convenient.

Other tools have similar functionality:
* [`npm logout`](https://docs.npmjs.com/cli/v7/commands/npm-logout)
* [`gem signout`](https://guides.rubygems.org/command-reference/#gem-signout)
* [`yarn logout`](https://classic.yarnpkg.com/en/docs/cli/logout)
* [`dart pub token remove`](https://dart.dev/tools/pub/cmd/pub-token)
* [`docker logout`](https://docs.docker.com/engine/reference/commandline/logout/)
* [`aws logout`](https://awscli.amazonaws.com/v2/documentation/api/latest/reference/sso/logout.html)
* [`gh auth logout`](https://cli.github.com/manual/gh_auth_logout)

(and probably many others)

I do not know of any concerns about the design or implementation of the command. Since I doubt we will discover anything new by keeping it unstable, I'd like to propose to stabilize it.

### PR DESCRIPTION
**Pull request by ehuss**
This stabilizes the `cargo logout` command.

Closes #11910
Closes #11884

### CODE REVIEW

**Comment by ehuss** [comment]
Marking as a draft pending #11949.

File: src/doc/man/cargo-logout.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+## DESCRIPTION
+
+This command will remove the API token from the local credential storage.
+Credentials are stored in `$CARGO_HOME/credentials.toml` where `$CARGO_HOME`
+defaults to `.cargo` in your home directory.
+
+If `--registry` is not specified, then the credentials for <https://crates.io/>
+will be removed.
+
```
Comment (jofas) [inline src/doc/man/cargo-logout.md:19]:
```suggestion
```

I think this should be removed if #11949 gets merged.

**Review by jofas — COMMENTED** [review]
If #11949 gets merged this sentence is wrong, because not the credentials of crates.io are removed but the ones of the default registry.

**Review by epage — APPROVED** [review]
With the caveat of the documentation pointed out by jofas

File: src/doc/man/cargo-logout.md (hunk 1 above)
Comment (epage) [reply src/doc/man/cargo-logout.md]: Looks like this is now resolved

**Comment by epage** [comment]
@ehuss any reason for this PR to still be marked as a Draft?

**Comment by ehuss** [comment]
This should be ready. I was just waiting for CI to be green to make sure I didn't miss anything.

**Comment by ehuss** [comment]
The documentation should be updated. I'm going to go ahead and move this forward (so I don't need to deal with updating the release version) assuming the above approval was a green light.  (Feel free to let me know if there are any changes wanted with the docs.)

@bors r=epage

**Pull request merged**